src/invariant/LedgerEntryIsValid.cpp

```diff
@@ -368,4 +370,42 @@ LedgerEntryIsValid::checkIsValid(LedgerEntry const& le,
 
     return {};
 }
+
+std::string
+LedgerEntryIsValid::checkIsValid(LiquidityPoolEntry const& lp,
+                                 uint32 version) const
+{
+    auto const cp = lp.body.constantProduct();
+    if (!isAssetValid(cp.params.assetA))
+    {
+        return "LiquidityPool assetA is invalid";
+    }
+    if (!isAssetValid(cp.params.assetB))
+    {
+        return "LiquidityPool assetB is invalid";
+    }
+    if (cp.params.fee != 30)
+    {
+        return "LiquidityPool fee is not 30 basis points";
+    }
+
+    // TODO: Invariants on the fields below can be improved
```

Comment:
Nitpick: don't leave TODOs in the code, better to resolve them or open issues

Reply:
Removed. I took note of this instead.